src/algorithms/edgegradient.jl

```diff
@@ -0,0 +1,18 @@
+struct EdgeGradient <: NeutralLandscapeMaker
+    direction::Float64
+    EdgeGradient(x::T) where {T <: Real} = new(mod(x, 360.0))
+end
+
+EdgeGradient() = EdgeGradient(360rand())
+
+"""
+    landscape(alg::EdgeGradient, s::Tuple{IT,IT}=(10,30)) where {IT <: Integer}
+
+Creates a landscape of size `s` following the planar gradient model. The
+additional arguments `kw...` are passed to the post-processing function, see the
+documentation of **TODO**. 
+"""
+function landscape(alg::EdgeGradient, s::Tuple{IT,IT}=(10,30); kw...) where {IT <: Integer}
```

Comment:
Could we have a function instead:
```julia
function landscape(alg, size; mask = Nothing, kw...)
  ret = _landscape(alg, size; kw...)
  isnothing(mask) || mask!(ret)
  rescale!(ret)
end
```
?

Reply:
would save a lot of the boilerplate in the python version

Reply:
@gottacatchenall ^

Reply:
i like this structure but i wonder if there is a case where a user would not want it rescaled---we could do something similar
as mask, ie

```
function landscape!(mat, alg, size; scale = true, mask = Nothing, kw...)
  _landscape!(mat!, alg, size; kw...)
  isnothing(mask) || mask!(ret)
  scale && rescale!(ret)
end
```

or something similar, although i'm not sure if there is a need for that use-case

Reply:
the issue with not rescaling is that it *does* still scale, but with network size - this is not ideal, I think

Reply:
> and then have the dispatching function be `_landscape!`?

This is so obvious... Yes!

Reply:
https://github.com/EcoJulia/NeutralLandscapes.jl/pull/2

Reply:
> the issue with not rescaling is that it _does_ still scale, but with network size - this is not ideal, I think

hmm i guess i'm thinking if you want different types of scaling (linear scaling on an arbitrary interval,  standardization, normalization, etc), you could use something like this to use an arbitrary scaling function

```
function landscape!(mat, alg, size; scaler = scale!, mask = Nothing, kw...)
  _landscape!(mat!, alg, size; kw...)
  isnothing(mask) || mask(ret)
  scaler(ret)
end
```

but either way works for me

Reply:
I'd think so - fidelity first. Scaling is also quite cheap I'd think